lib/commentOnClosedIssue.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+var Cesium = require('cesium');
+var Promise = require('bluebird');
+var rp = require('request-promise');
+
+var getUniqueMatch = require('./getUniqueMatch');
+
+var defined = Cesium.defined;
+
+/** Get comments -> regex search -> post comment
+ *
+ * @param {String} jsonResponse GitHub Response
+ * @param {Object} headers Request headers
+ * @return {Promise<http.IncomingMessage>} Response
+ */
+var commentOnClosedIssue = function(jsonResponse, headers) {
+    if (!defined(jsonResponse) || !defined(headers)) {
+        return Promise.reject('jsonResponse is undefined');
+    }
+    var commentsUrl = jsonResponse.issue.comments_url;
```

Comment:
That's a good strategy, but what I would do is have `commentOnClosedIssue` do nothing but extract out the `comemnts_url` and then pas it to a `commentOnClosedIssue._implemtnation` function.  This still keeps everything in the file but decouples the function, making things easier to test.  We do this in Composer all over the place, see https://github.com/AnalyticalGraphicsInc/agi-cesium-cloud/blob/master/routes/AssetServices.js#L678 for an example.